packages/jellyfish-transaction/src/script/defi/dftx_governance.ts

```diff
@@ -0,0 +1,91 @@
+import BigNumber from 'bignumber.js'
+import { SmartBuffer } from 'smart-buffer'
+import { readBigNumberUInt64, writeBigNumberUInt64 } from '../../buffer/buffer_bignumber'
+import { BufferComposer, ComposableBuffer } from '../../buffer/buffer_composer'
+
+// Disabling no-return-assign makes the code cleaner with the setter and getter */
+/* eslint-disable no-return-assign */
+
+export interface LiqPoolSplit {
+  tokenId: number // -------------------| 4 bytes unsigned
+  value: BigNumber // ------------------| 8 bytes unsigned
+}
+
+export class CLiqPoolSplit extends ComposableBuffer<LiqPoolSplit> {
+  composers (lps: LiqPoolSplit): BufferComposer[] {
+    return [
+      ComposableBuffer.uInt32(() => lps.tokenId, v => lps.tokenId = v),
+      ComposableBuffer.satoshiAsBigNumber(() => lps.value, v => lps.value = v)
+    ]
+  }
+}
+
+export interface GovernanceVar {
+  key: string // -----------------------| VarUInt{1-9 bytes}
+  value: BigNumber | LiqPoolSplit[] // -| VarUInt{8 OR 1 + n * 12 bytes}, case LiqPoolSplit: first byte = array len
+}
+
+export class CGovernanceVar extends ComposableBuffer<GovernanceVar> {
+  composers (gv: GovernanceVar): BufferComposer[] {
+    return [
+      ComposableBuffer.varUIntUtf8BE(() => gv.key, v => gv.key = v),
+      {
+        fromBuffer: (buffer: SmartBuffer): void => {
+          if (gv.key === 'LP_DAILY_DFI_REWARD') {
+            gv.value = readBigNumberUInt64(buffer).div('1e8')
+          } else if (gv.key === 'LP_SPLITS') {
+            gv.value = []
+            const configLen = buffer.readUInt8()
+            for (let i = 0; i < configLen; i++) {
+              gv.value.push(new CLiqPoolSplit(buffer).toObject())
+            }
+          } else {
+            throw new Error(`Unrecognized Governance Variable type: ${gv.key}`)
```

Comment:
Any idea how to handle this better, if the decoder (somehow) not updated for other new governance var in future?

Reply:
@fuxingloh not sure you have better design for it??
```
// structure
data: {
  governanceVars: [
    { key: 'LP_SPLITS', value: [{
      tokenId: 2, value: BigNumber('2'),
      tokenId: 4, value: BigNumber('8'),
    }]},
    { key: 'LP_DAILY_DFI_REWARD', value: new BigNumber('1')},
  ]
}
```